jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -175,6 +175,12 @@ def _on_glue_y_display_unit_changed(self, y_unit_str):
         if check_if_unit_is_per_solid_angle(y_unit_str):
             flux_choices = create_flux_equivalencies_list(y_unit * u.sr, x_unit)
             self.flux_unit.choices = flux_choices
+            flux_unit = str(y_unit * u.sr)
+            # We need to set the angle_unit before triggering _on_flux_unit_changed
+            self.angle_unit.choices = create_angle_equivalencies_list(y_unit)
```

Comment:
why does the angle unit need to be set here if it's being set below after `if self.app.data_collection[0]:`

Reply:
Because setting `self.flux_unit.selected` below triggers `_on_flux_unit_changed`, which uses the angle.

Reply:
I guess the comment could have been a little clearer.

Reply:
I added some more detail to the comment above this.

Reply:
makes sense, thanks!